File: benchmark-java-node-php-python/java-webflux/monolith/src/main/java/com/benchmark/monolith/repository/BonusRepository.java
package com.benchmark.monolith.repository;

import com.benchmark.monolith.domain.Bonus;
import org.springframework.data.repository.reactive.ReactiveCrudRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface BonusRepository extends ReactiveCrudRepository<Bonus, Long> {
}

File: benchmark-java-node-php-python/golang/monolith/internal/repository/bonus_repository.go
package repository

import (
	"monolith/internal/domain"
	"gorm.io/gorm"
)

type BonusRepository interface {
	Create(bonus *domain.Bonus) error
	FindByID(id string) (*domain.Bonus, error)
}

type bonusRepository struct {
	db *gorm.DB
}

func NewBonusRepository(db *gorm.DB) BonusRepository {
	return &bonusRepository{db}
}

func (r *bonusRepository) Create(bonus *domain.Bonus) error {
	return r.db.Create(bonus).Error
}

func (r *bonusRepository) FindByID(id string) (*domain.Bonus, error) {
	var bonus domain.Bonus
	if err := r.db.First(&bonus, id).Error; err != nil {
		return nil, err
	}
	return &bonus, nil
}
